sotodlib/coords/helpers.py

```diff
@@ -228,11 +243,33 @@ def get_wcs_kernel(proj, ra, dec, res):
     Returns a WCS object that captures the requested pixelization.
 
     """
-    assert np.isscalar(res)  # This ain't enlib.
-    _, wcs = enmap.geometry(np.array((dec, ra)), shape=(1,1), proj=proj,
-                            res=(res, -res))
+    if len(proj) > 3:
+        m = re.match('(?P<proj>car|tan)-(?P<res>[0-9]*.?[0-9]*)(?P<unit>arcsec|arcmin|deg)', proj)
+        proj = m['proj']
+        res = float(m['res'])
+        unit = m['unit']
+
+        # Convert to radians
+        res = res * np.pi/180.
+        if unit == 'arcmin':
+            res /= 60.
+        if unit == 'arcsec':
+            res /= 3600.
+
+        if unit == 'deg' and res > 10:
+            raise ValueError("Beam is too big")
+
+        _, wcs = enmap.geometry(np.array((0, 0)), shape=(1, 1), proj=proj,
+                                res=(res, -res))
+
+    else:
+        assert np.isscalar(res)  # This ain't enlib.
+        _, wcs = enmap.geometry(np.array((dec, ra)), shape=(1, 1), proj=proj,
+                                res=(res, -res))
+
     return wcs
```

Comment:
Before returning, tweak reference pixel offset to 0.5.  But only for CAR projection.  This makes it Fejer1 compatible (good for SHTs).
```
if proj == 'car':
    wcs.wcs.crpix = [1.0, 0.5]
```